rust/sample_connector_scaffolding/src/main.rs

```diff
@@ -0,0 +1,699 @@
+
+        // For this example, we will assume that the asset specification is OK, see below for how to handle a bad update.
+        break local_asset_specification;
+
+        // // If the asset specification is not OK await for a new one
+        // let _ = asset_status_reporter_watcher_rx.changed().await;
+        // // Update the status reporter for the new asset specification
+        // asset_status_reporter_tx = asset_status_reporter_watcher_rx.borrow_and_update().clone();
+    };
+
+    // This boolean tracks if the dataset and asset are ready to be sampled.
+    let mut is_dataset_asset_ready = true;
+    // This boolean tracks if the status for the dataset has been reported.
+    let mut is_dataset_reported = false;
+    // Extract the dataset definition from the dataset client
+    #[allow(clippy::never_loop)]
+    let mut local_dataset_definition = loop {
+        let local_dataset_definition = dataset_client.dataset_definition().clone();
+
+        // TODO: Verify the dataset definition is OK
+
+        // For this example, we will assume that the dataset definition is OK, see below for how to handle a bad update.
+        break local_dataset_definition;
+
+        // // If the dataset definition is not OK await for a new one
+        // match dataset_client.recv_notification().await {
+        //     DatasetNotification::Updated => {
+        //         log::info!("{dataset_log_identifier} Dataset definition updated");
+        //     }
+        //     DatasetNotification::Deleted => {
+        //         log::info!("{dataset_log_identifier} Dataset deleted, ending dataset handler");
+        //         return;
+        //     }
+        //     DatasetNotification::UpdatedInvalid => {
+        //         log::info!("{dataset_log_identifier} Dataset definition invalid, waiting for a valid update");
+        //         loop {
+        //             // Wait for a valid update
+        //             match dataset_client.recv_notification().await {
+        //                 DatasetNotification::Updated => {
+        //                     log::info!("{dataset_log_identifier} Dataset definition updated");
+        //                     break;
+        //                 }
+        //                 DatasetNotification::Deleted => {
+        //                     log::info!("{dataset_log_identifier} Dataset deleted, ending dataset handler");
+        //                     return;
+        //                 }
+        //                 DatasetNotification::UpdatedInvalid => {
+        //                     log::info!("{dataset_log_identifier} Dataset definition invalid, waiting for a valid update");
+        //                 }
+        //             }
+        //         }
+        //     }
+        // }
+    };
+
+    // This variable keeps track of the latest reported schema.
+    let mut current_schema = None;
+
+    let mut timer = tokio::time::interval(DEFAULT_SAMPLING_INTERVAL);
```

Comment:
```suggestion
    // this could be configured in the dataset_configuration instead if desired
    let mut timer = tokio::time::interval(DEFAULT_SAMPLING_INTERVAL);
```